Interesting. Builds are all working locally just fine with this change and the build is succeeding in CI in that the debug and release are combining together. Yet somehow I've broken the loading of the Roslyn compiler. Lots of the following:

> C:\Users\vsagent\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta1-19606-04\tasks\netcoreapp2.1\Microsoft.CSharp.Core.targets(59,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Users\vsagent\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta1-19606-04\build\..\tasks\netcoreapp2.1\Microsoft.Build.Tasks.CodeAnalysis.dll. Assembly with same name is already loaded Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [F:\workspace\_work\1\s\src\libraries\System.Security.Principal.Windows\ref\System.Security.Principal.Windows.csproj]
C:\Users\vsagent\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta1-19606-04\tasks\netcoreapp2.1\Microsoft.CSharp.Core.targets(59,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Users\vsagent\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta1-19606-04\build\..\tasks\netcoreapp2.1\Microsoft.Build.Tasks.CodeAnalysis.dll. Assembly with same name is already loaded Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [F:\workspace\_work\1\s\src\libraries\System.ComponentModel.Composition\ref\System.ComponentModel.Composition.csproj]

Generally that means we're attempting to load the Roslyn build task from two different locations during the build. Unsure how my changes could impact where Roslyn is loading from. Hoping I can reproduce this on my machine, or at least the binary log will give me more insight into where we do the different loads so I can track it down. There is no explicit `<UsingTask ...>` in the runtime code though which should lead to this.

--- a/eng/pipelines/common/global-build-job.yml
+++ b/eng/pipelines/common/global-build-job.yml
@@ -36,7 +36,7 @@ jobs:
     - template: /eng/pipelines/common/clone-checkout-bundle-step.yml
 
     # Build
-    - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -c ${{ parameters.buildConfig }} -runtimeConfiguration ${{ parameters.runtimeConfig }} -arch ${{ parameters.archType }} -bl
+    - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -c ${{ parameters.buildConfig }} -runtimeConfiguration ${{ parameters.runtimeConfig }} -arch ${{ parameters.archType }} -bl -ci
       displayName: Build product
 
     - task: PublishBuildArtifacts@1

--- a/eng/pipelines/common/global-build-job.yml
+++ b/eng/pipelines/common/global-build-job.yml
@@ -36,7 +36,7 @@ jobs:
     - template: /eng/pipelines/common/clone-checkout-bundle-step.yml
 
     # Build
-    - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -c ${{ parameters.buildConfig }} -runtimeConfiguration ${{ parameters.runtimeConfig }} -arch ${{ parameters.archType }} -bl -ci
+    - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -c ${{ parameters.buildConfig }} -runtimeConfiguration ${{ parameters.runtimeConfig }} -arch ${{ parameters.archType }} -ci
       displayName: Build product
 
     - task: PublishBuildArtifacts@1

--- a/eng/pipelines/common/global-build-job.yml
+++ b/eng/pipelines/common/global-build-job.yml
@@ -36,5 +36,14 @@ jobs:
     - template: /eng/pipelines/common/clone-checkout-bundle-step.yml
 
     # Build
-    - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -c ${{ parameters.buildConfig }} -runtimeConfiguration ${{ parameters.runtimeConfig }} -arch ${{ parameters.archType }}
+    - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -c ${{ parameters.buildConfig }} -runtimeConfiguration ${{ parameters.runtimeConfig }} -arch ${{ parameters.archType }} -bl
       displayName: Build product
+
+    - task: PublishBuildArtifacts@1
+      displayName: Publish Logs
+      inputs:
+        PathtoPublish: '$(Build.SourcesDirectory)/artifacts/log/'
+        PublishLocation: Container
+        ArtifactName: $(Agent.Os)_PublishBuildAssets
+      continueOnError: true
+      condition: always()

--- a/src/coreclr/build.cmd
+++ b/src/coreclr/build.cmd
@@ -386,7 +386,8 @@ if %__RestoreOptData% EQU 1 (
     echo %__MsgPrefix%Restoring the OptimizationData Package
     powershell -NoProfile -ExecutionPolicy ByPass -NoLogo -File "%__RepoRootDir%\eng\common\msbuild.ps1" /clp:nosummary %__ArcadeScriptArgs%^
         %OptDataProjectFilePath% /t:Restore^
-        %__CommonMSBuildArgs% %__UnprocessedBuildArgs%
+        %__CommonMSBuildArgs% %__UnprocessedBuildArgs%^
+        /nodereuse:false
     if not !errorlevel! == 0 (
         echo %__ErrMsgPrefix%%__MsgPrefix%Error: Failed to restore the optimization data package.
         set __exitCode=!errorlevel!

--- a/src/coreclr/build.sh
+++ b/src/coreclr/build.sh
@@ -53,7 +53,8 @@ restore_optdata()
         echo "Restoring the OptimizationData package"
         "$__RepoRootDir/eng/common/msbuild.sh" /clp:nosummary $__ArcadeScriptArgs \
                                                $OptDataProjectFilePath /t:Restore /m \
-                                               $__CommonMSBuildArgs $__UnprocessedBuildArgs
+                                               $__CommonMSBuildArgs $__UnprocessedBuildArgs \
+                                               /nodereuse:false
         local exit_code="$?"
         if [[ "$exit_code" != 0 ]]; then
             echo "${__ErrMsgPrefix}Failed to restore the optimization data package."

